addons/udes_stock/tests/test_stock_picking.py

```diff
@@ -0,0 +1,289 @@
+        locations = self.test_goodsout_locations | self.out_location
+        self.assertEqual(self.test_picking_pick._get_child_dest_locations(), locations)
+
+    def test03_get_child_locations_simple_success_with_extra_domain(self):
+        """ Get child locations - with extra domain"""
+        aux_domain = [('name', '=', self.test_goodsout_location_01.name)]
+        self.assertEqual(
+            self.test_picking_pick._get_child_dest_locations(aux_domain=aux_domain),
+            self.test_goodsout_location_01,
+        )
+
+    def test04_get_child_locations_failure_with_extra_domain(self):
+        """ Return no child locations - with extra domain"""
+        aux_domain = [('name', '=', 'Not a location')]
+        self.assertFalse(self.test_picking_pick._get_child_dest_locations(aux_domain=aux_domain))
+
+    def test05_create_picking_no_moves(self):
+        """ Create a picking from picking type but have no moves created as no proudts_info given """
+        pick = self.Picking.create_picking(picking_type=self.picking_type_pick)
+        # Default names for pick location
+        self.assertEqual(pick.location_id, self.stock_location)
+        self.assertEqual(pick.location_dest_id, self.out_location)
+        self.assertEqual(len(pick.move_lines), 0)
+        # State is in draft
+        self.assertEqual(pick.state, 'draft')
+        # Check batch not created
+        self.assertFalse(pick.batch_id)
+
+    def test06_create_picking_success_simple(self):
+        """ Create a picking from picking type """
+        self.create_quant(self.apple.id, self.test_stock_location_01.id, 50)
+        self.create_quant(self.banana.id, self.test_stock_location_01.id, 50)
+        products = (self.apple | self.banana)
+        pick = self.Picking.create_picking(
+            picking_type=self.picking_type_pick,
+            products_info=[{'product': self.apple, 'qty': 2}, {'product': self.banana, 'qty': 4},],
+        )
+        # Default names for pick location
+        self.assertEqual(pick.location_id, self.stock_location)
+        self.assertEqual(pick.location_dest_id, self.out_location)
+        self.assertEqual(pick.move_lines.product_id, products)
+        # State is in draft
+        self.assertEqual(pick.state, 'draft')
+        # Check batch not created
+        self.assertFalse(pick.batch_id)
+
+    def test07_create_picking_success_custom_locations(self):
+        """ Create a picking with non-default locations and confirm"""
+        pick = self.Picking.create_picking(
+            picking_type=self.picking_type_pick,
+            products_info=[{'product': self.apple, 'qty': 2}, {'product': self.banana, 'qty': 4},],
+            location_id=self.test_stock_location_01.id,
+            location_dest_id=self.test_goodsout_location_01.id,
+            confirm=True,
+            assign=False,
+            priority='0',
+            create_batch=True,
+        )
+        products = (self.apple | self.banana)
```

Comment:
nit: for consistency I would have this line alway before or after create_picking(), since atm is different from one test to another.